web/server/api/endpoints/files.py

```diff
@@ -162,18 +187,16 @@ def walk_path(
     )
 
 
-def _get_file_with_content(
-    path: Path,
-    settings: Settings,
-) -> models.File:
+def _get_file_with_content(file_path: Path, path: str) -> models.File:
```

Comment:
The parameters are very similar. Is it because you don't want to pass in the settings object? Then how about `project_path: Path` and `relative_path: str`?